src/TableFunctions/ITableFunctionDataLake.h

```diff
@@ -34,7 +34,7 @@ class ITableFunctionDataLake : public TableFunction
             columns = parseColumnsListFromString(TableFunction::configuration.structure, context);
 
         StoragePtr storage = Storage::create(
-            TableFunction::configuration, context, StorageID(TableFunction::getDatabaseName(), table_name),
+            TableFunction::configuration, context, false, StorageID(TableFunction::getDatabaseName(), table_name),
```

Comment:
What if table was created using `CREATE TABLE ... AS iceberg(...)`. Will we use table function during attach query?

Reply:
Good point, though added a test in [a4082f5](https://github.com/ClickHouse/ClickHouse/pull/59080/commits/a4082f5b971b6da51bcdc65fca26b21b4b429c84) and it seems to work...

Reply:
Also made a test locally, and seems like in case of `CREATE TABLE ... AS iceberg(...)` during server startup we execute table function lazily on first request, so we don't actually attach Iceberg table